pycbc/waveform/generator.py

```diff
@@ -525,6 +526,8 @@ def generate(self, *args):
         rfparams = dict([(param, self.current_params[param])
             for param in self.rframe_generator.variable_args])
         hp, hc = self.rframe_generator.generate_from_kwargs(**rfparams)
+        hp = filter.make_frequency_series(hp)
+        hc = filter.make_frequency_series(hc)
         hp._epoch = hc._epoch = self._epoch
```

Comment:
This isn't enough conditioning. Three things need to happen: 1.) apply tapering to the time domain waveform; 2.) resize the TD array so that it gives the correct delta_f when FFT'd to the frequency domain; 3.) set the `_epoch` so that the waveform will be shifted such that it's coalescence time will be at the start of the array.

For tapering, I'd add something to `TDomainCBCGenerator` that will apply tapering if a `taper` argument is in the current params. The easiest way to do that is to just define a `postgenerate` function under `TDomainCBCGenerator` that, if a `"taper"` argument is in `self.current_params`, it applies the taper then returns the waveforms. That'll override `BaseGenerator`'s `postgenerate`, which gets called inside the generate decorator.

For resizing, just do that at line 528. You'll need to make it the length of the (data - 1)*2 long. Likewise, for setting the epoch, all time-domain waveforms should return with the epoch being -(the amount of time to the peak). So set `hp._epoch` and `hc._epoch` to be `self._epoch + hp._epoch`. The time shift that happens in `apply_fd_timeshift` will then automatically take care of the difference. Finally, after doing those two things, convert to the frequency series.